src/chromo/models/epos.py

```diff
@@ -23,6 +23,50 @@ def _get_impact_parameter(self):
     def _get_n_wounded(self):
         return int(self._lib.cevt.npjevt), int(self._lib.cevt.ntgevt)
 
+    def _history_zero_indexing(self):
+        # Beam particles are [-1, -1], but indexing starts from 1
+        # adjust to [0, 0]
+        condition = (self.mothers == [-1, -1]).all(axis=1)
+        self.mothers[condition] = [0, 0]
+
+        # Adjust to zero-base indexing
+        self.mothers = self.mothers - 1
+
+        # Set [i, i] to [i, -1]
+        condition = self.mothers[:, 0] == self.mothers[:, 1]
+        self.mothers[condition, 1] = -1
+
+        self.daughters = self.daughters - 1
+        condition = self.daughters[:, 0] == self.daughters[:, 1]
+        self.daughters[condition, 1] = -1
+
+    def _repair_initial_beam(self):
+        # projectile and target are switched!!!
```

Comment:
The order of projectile and target can not be switched, since soem configurations are asymmetric. Something else needs to be done here.

Reply:
The code is changed to take into account that both projectile and target could be nucleus. No swaping of projectile and target occurs now. The problem why the swap was introduced is fixed following way.

The problem occured when some particles were attached to the projectile and several wounded nucleons. When target nucleous inserted between them, it created a configuration which hepmc cannot process. I attach all such wounded nucleons only to projectile, whenever such situation occurs.